### PR TITLE
[17.0][MIG] account_loan: Migration to 17.0

### PR DESCRIPTION
**Pull request by victoralmau**
Supersedes https://github.com/OCA/account-financial-tools/pull/1925

Migration to 17.0

Changes done:
- [x] Purge translation files
- [x] Remove unnecessary `readonly=true` to some fields (fields defining the readonly in the view and/or related)
- [x] Remove unnecesary `default=false` to boolean fields (redundant, is the default behavior)
- [x] Remove unnecessary _or_ text from views
- [x] Improve multi-company rule + migration script
- [x] Improve tests

Please @pedrobaeza and @pilarvargas-tecnativa can you review it?

@Tecnativa TT51502

### CODE REVIEW

File: account_loan/security/account_loan_security.xml [hunk 1]
```diff
@@ -4,17 +4,13 @@
         <field name="name">Account loan multi-company</field>
         <field ref="model_account_loan" name="model_id" />
         <field eval="True" name="global" />
-        <field name="domain_force">
-            ['|',('company_id','=',False),('company_id','in',company_ids)]
-        </field>
+        <field name="domain_force">[('company_id', 'in', company_ids + [False])]</field>
```
Comment (pedrobaeza) [inline account_loan/security/account_loan_security.xml:7]: Add migration script for this

File: account_loan/models/account_loan.py [hunk 2]
```diff
@@ -29,22 +29,19 @@ def _default_company(self):
         required=True,
         readonly=True,
```
Comment (pilarvargas-tecnativa) [inline account_loan/models/account_loan.py]: This readonly is necessary?

Example of the migration guide:
https://github.com/odoo/odoo/pull/104741/files#diff-1e3bd6be3bfb83a37ec9fb800ce8b1c95afe0be90ff792874ae7299c320a2f6eL162-L163

File: account_loan/views/account_loan_view.xml [hunk 3]
```diff
@@ -94,34 +94,41 @@
...
                         <group>
-                            <field name="company_id" options="{'no_create': True}" />
-                            <field name="loan_type" />
-                            <field name="loan_amount" />
+                            <field
+                                name="company_id"
+                                options="{'no_create': True}"
+                                readonly="state != 'draft'"
+                            />
+                            <field name="loan_type" readonly="state != 'draft'" />
```
Comment (pilarvargas-tecnativa) [inline account_loan/views/account_loan_view.xml:108]: This is related to the above, remove the state from the field and add this in the view. So perhaps the readonly would not be necessary in the field.
There are several fields where this field is being made and the readonly has not been removed.

Comment (victoralmau) [reply account_loan/views/account_loan_view.xml:108]: Right, thanks, it's already changed

File: account_loan/views/account_loan_view.xml [hunk 4]
```diff
@@ -138,38 +145,54 @@
...
+                                    />
                                     <field name="currency_id" invisible="1" />
                                 </group>
                             </group>
                         </page>
                         <page
                             string="Leasing"
                             id="leasing"
-                            attrs="{'invisible': [('is_leasing', '=', False)]}"
+                            invisible="is_leasing == False"
```
Comment (pilarvargas-tecnativa) [inline account_loan/views/account_loan_view.xml]: "not is_leasing"

File: account_loan/views/account_loan_view.xml [hunk 5]
```diff
@@ -138,38 +145,54 @@
...
                         >
                             <group>
                                 <group>
                                     <field
                                         name="leased_asset_account_id"
-                                        attrs="{'required': [('is_leasing', '=', True)]}"
+                                        required="is_leasing == True"
```
Comment (pilarvargas-tecnativa) [inline account_loan/views/account_loan_view.xml]: "is_leasing"

File: account_loan/views/account_loan_view.xml [hunk 6]
```diff
@@ -138,38 +145,54 @@
...
+                                        name="residual_amount"
+                                        readonly="state != 'draft'"
                                     />
-                                    <field name="residual_amount" />
                                 </group>
                                 <group>
                                     <field
                                         name="product_id"
-                                        attrs="{'required': [('is_leasing', '=', True)]}"
+                                        required="is_leasing == True"
```
Comment (pilarvargas-tecnativa) [inline account_loan/views/account_loan_view.xml]: The same

File: account_loan/views/account_loan_view.xml [hunk 7]
```diff
@@ -138,38 +145,54 @@
...
                                     />
                                     <field
                                         name="interests_product_id"
-                                        attrs="{'required': [('is_leasing', '=', True)]}"
+                                        required="is_leasing == True"
```
Comment (pilarvargas-tecnativa) [inline account_loan/views/account_loan_view.xml]: And here.

In account_move_view.xml too

**Review by pilarvargas-tecnativa — APPROVED** [review]

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge nobump

**Pull request merged**